atlante/server/server.go

```diff
@@ -0,0 +1,219 @@
+o888o   o888o `Y8bod8P' `Y888""8o `Y8bod88P" `Y8bod8P' d888b          `Y8bood8P'   `Y8bod8P'     888 `Y8bod8P' `Y8bod8P'   "888"
+                                                                                                 888
+                                                                                             .o. 88P
+																							 `Y888P
+*/
+
+func (h Headers) setHeaders(w http.ResponseWriter) {
+	// add CORS headers
+	for name, val := range DefaultCORSHeaders {
+		if val == "" {
+			log.Warnf("default CORS header (%v) has no value", name)
+		}
+		w.Header().Set(name, val)
+	}
+
+	// set user defined headers
+	for name, val := range Headers {
+		if val == "" {
+			log.Warnf("header (%v) has no value", name)
+		}
+		w.Header().Set(name, val)
+	}
+}
+
+// corsHandler
+func (h Headers) corsHandler(w http.ResponseWriter, _ *http.Request, _ map[string]string) {
+	h.setHeaders(w)
+}
+
+/*
+ .oooooo..o                                                          .oooooo.    .o8           o8o                         .
+d8P'    `Y8                                                         d8P'  `Y8b  "888           `"'                       .o8
+Y88bo.       .ooooo.  oooo d8b oooo    ooo  .ooooo.  oooo d8b      888      888  888oooo.     oooo  .ooooo.   .ooooo.  .o888oo
+ `"Y8888o.  d88' `88b `888""8P  `88.  .8'  d88' `88b `888""8P      888      888  d88' `88b    `888 d88' `88b d88' `"Y8   888
+     `"Y88b 888ooo888  888       `88..8'   888ooo888  888          888      888  888   888     888 888ooo888 888         888
+oo     .d8P 888    .o  888        `888'    888    .o  888          `88b    d88'  888   888     888 888    .o 888   .o8   888 .
+8""88888P'  `Y8bod8P' d888b        `8'     `Y8bod8P' d888b          `Y8bood8P'   `Y8bod8P'     888 `Y8bod8P' `Y8bod8P'   "888"
+                                                                                               888
+                                                                                           .o. 88P
+																						   `Y888P
+*/
+
+// GridInfoHandler will writer to the writer information about the requested grid. The params should contain a `sheet_name` and either an `mgdid` entry
+// or a `lat` and `lng` entries.
+// if an non-empty `mgdid` string is there it will be used to attempt to retrieve grid information based on that value.
+// if a `lat` and `lng` entries is given the mgdid grid containing that point will be used instead.
+// if both are provided, then the mdgid will be used, and the lat/lng keys will be ignored.
+// if neigher are provided or if the value is bad then a 400 status code will be returned.
+// Requires
+func (s *Server) GridInfoHandler(w http.ResponseWriter, request *http.Request, params map[string]string) {
+
+	var (
+		mdgid grids.MDGID
+		grid  *grid.Grid
+		err   error
+	)
+	sheetName, ok := params[ParamsKeySheetName]
+	if !ok {
+		// We need a sheetnumber.
+		w.Header["x-issue"] = "missing sheet name"
```

Comment:
is the strategy to use the `x-issue` header instead of a response payload that the client will use for communicating errors to the user?